### PR TITLE
Operator can add and remove ES nodes

### PR DESCRIPTION
**Pull request by mburke5678**
https://jira.coreos.com/browse/OSDOCS-228

### CODE REVIEW

**Comment by mburke5678** [comment]
@jcantrill PTAL. This topic is based on https://jira.coreos.com/browse/OSDOCS-228 and the associated test cases. Please let me know if this is appropriate for 4.0. 
Any additional use cases for scaling an ES cluster would be helpful, too.

**Comment by mburke5678** [comment]
@jcantrill PTAL. This topic is based on https://jira.coreos.com/browse/OSDOCS-228 and the associated test cases. Please let me know if this is appropriate for 4.0.
Any additional use cases for scaling an ES cluster would be helpful, too.

File: modules/efk-logging-elasticsearch-add-remove.adoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// Module included in the following assemblies:
+//
+// * logging/efk-logging-elasticsearch.adoc
+
+[id='efk-logging-elasticsearch-add-remove_{context}']
+= Changing the number of Elasticsearch nodes
+
+You can scale the number of nodes in Elasticsearch.
+
+For example, if you want to increase redundancy, and use the `FulRedundancy` or `MultipleRedundancy` policy, you can add additional data nodes to increase the number of shard replicas in your cluster.
```
Comment (jcantrill) [inline modules/efk-logging-elasticsearch-add-remove.adoc]: ..."FullRedundancy"...

File: modules/efk-logging-elasticsearch-add-remove.adoc [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+// * logging/efk-logging-elasticsearch.adoc
+
+[id='efk-logging-elasticsearch-add-remove_{context}']
+= Changing the number of Elasticsearch nodes
+
+You can scale the number of nodes in Elasticsearch.
+
+For example, if you want to increase redundancy, and use the `FulRedundancy` or `MultipleRedundancy` policy, you can add additional data nodes to increase the number of shard replicas in your cluster.  
+
+A {product-title} Elasticsearch cluster can contain three node types, or roles:
```
Comment (jcantrill) [inline modules/efk-logging-elasticsearch-add-remove.adoc]: @ewolinetz Thoughts here?  I think we should exclude it all together as there is no way to modify via CL and I'm not certain how well the EO at this time supports modifying should we advise to go unmanged

File: modules/efk-logging-elasticsearch-add-remove.adoc [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+* *Master*. A master node performs actions across the cluster, such as creating or deleting an index, tracking which nodes are part of the cluster, and deciding which shards to allocate to which nodes. The master node must have access to the the *_data/_* directory. You can create 1 or 3 master nodes.
+* *Data*. A data node is where {product-title} stores Elasticsearch shards.
+* *Client*. A client node routes requests, handles the Elasticsearch reduce phase, and distributes bulk indexing. 
+
+A node can have any combination of these roles.
+
+.Procedure
+
+. To add a node, edit the cluster logging Custom Resource Definition (CR), by default *_../hack/cr.yaml_* to add a number of modes of a specific type:
```
Comment (jcantrill) [inline modules/efk-logging-elasticsearch-add-remove.adoc]: I would remove reference to the 'hack/cr.yaml' as this is not necessarily representative of the default. The one in the CSV is probably more accurate if we need to provide anything: https://github.com/operator-framework/community-operators/blob/master/community-operators/cluster-logging/clusterlogging.v0.0.1.clusterserviceversion.yaml#L16

File: modules/efk-logging-elasticsearch-add-remove.adoc [hunk 4]
```diff
@@ -0,0 +1,118 @@
+// Module included in the following assemblies:
```
Comment (jcantrill) [inline modules/efk-logging-elasticsearch-add-remove.adoc]: @ewolinetz Looking at this section in detail, I have reservations about describing, in detail, how to modify your ES cluster as it all needs to occur in "unmanaged' mode.  How far down this rabbit hole do we really want to go?

File: modules/efk-logging-elasticsearch-add-remove.adoc (hunk 3 above)
Comment (mburke5678) [reply modules/efk-logging-elasticsearch-add-remove.adoc]: @jcantrill What is the name of the cluster logging CRD that the user should edit? I reference editing the CRD elsewhere in other PRs. Is _clusterlogging.v0.0.1.clusterserviceversion.yaml_ at your link the CRD??

Comment (jcantrill) [reply modules/efk-logging-elasticsearch-add-remove.adoc]: They should never edit the CRD as this is the schema.  They will edit their CR which could be named anything they like.

**Comment by mburke5678** [comment]
Merging at the request of Jeff Cantrill to facilitate logging docs review.

File: modules/efk-logging-elasticsearch-add-remove.adoc (hunk 2 above)
Comment (ewolinetz) [reply modules/efk-logging-elasticsearch-add-remove.adoc]: @mburke5678 +1 on this, we don't want to expose this level of detail to customers. currently it is possible for them to control that, but we intend to take that away and have the operator make these scaling/topology decisions for customers.

**Pull request merged**